File: ios/Classes/style/AppInboxStyle.swift
//
//  AppInboxStyle.swift
//  exponea
//
//  Created by Adam Mihalik on 02/03/2023.
//

import Foundation

class AppInboxStyle {
    var appInboxButton: ButtonStyle?
    var detailView: DetailViewStyle?
    var listView: ListScreenStyle?

    init(appInboxButton: ButtonStyle? = nil, detailView: DetailViewStyle? = nil, listView: ListScreenStyle? = nil) {
        self.appInboxButton = appInboxButton
        self.detailView = detailView
        self.listView = listView
    }
}

File: lib/src/data/model/app_inbox_style.dart

import 'package:exponea/exponea.dart';

import '../util/Codable.dart';

class AppInboxStyle extends Encodable {
  final SimpleButtonStyle? appInboxButton;
  final DetailViewStyle? detailView;
  final ListScreenStyle? listView;

  AppInboxStyle({this.appInboxButton, this.detailView, this.listView});

  @override
  String toString() {
    return 'AppInboxStyle{appInboxButton: $appInboxButton, detailView: $detailView, listView: $listView}';
  }

  @override
  Map<String, dynamic> encode() {
    return {
      'appInboxButton': appInboxButton?.encodeClean(),
      'detailView': detailView?.encodeClean(),
      'listView': listView?.encodeClean()
    };
  }

}
